pages/en/zkapps/advanced-concepts.mdx

```diff
@@ -10,6 +10,126 @@ Build onto what you already learned regarding zkApps and SnarkyJS
 
 This page explores additional topics that require some familiarity with building zkApps, SnarkyJS, and crytography.
 
+## Using on-chain values
+
+We already saw how you can access the current <DocLink copy="on-chain state" url="/zkapps/how-to-write-a-zkapp/#reading-state" /> of a zkApp account. Similarly, you can access many other on-chain values in a zkApp.
+
+Just to give you an idea, here are two possible use cases:
+
+- Say you want to let users vote on a proposal, but only within a specific timespan. To do that, you can make your zkApp require that the current timestamp lies in a certain range.
+- In DeFi, you might need to compute amounts relative to a balance. For example, paying a yield of `0.001` times the account balance requires you to get the current on-chain balance.
+
+We group on-chain values in two categories:
+
+- **Network**: includes the current timestamp, block height, total Mina in circulation and other network state
+- **Account**: includes fields and properties of the zkApp account, such as balance, nonce and delegate
+
+The subfields of both categories are are accessible on `this.network` and `this.account` in your smart contract.
+For example, the timestamp is on `this.network.timestamp`, and it has four methods on it:
+
+```ts
+this.network.timestamp.get();
+this.network.timestamp.assertEquals(timestamp);
+this.network.timestamp.assertBetween(lower, upper);
+```
+
+With two of them you are already familiar: On-chain state has the same `get` and `assertEquals` methods. `assertBetween` gives you even more power: it allows you to make assert that the timestamp is between `lower` and `upper` (inclusive).
+
+### Example: restricting timestamps
+
+Let's see how `assertBetween` can be used in the voting example mentioned earlier. Assume we want to allow voting throughout September 2022. Timestamps are represented as a `UInt64` and counted in epoch milliseconds; the JS `Date` object makes it easy for us to convert to this representation. In the simplest case, our zkApp could just hard-code the dates:
+
+```ts
+let startDate = UInt64.from(Date.UTC(2022, 9, 1));
+let endDate = UInt64.from(Date.UTC(2022, 10, 1));
+
+class VotingApp extends SmartContract {
+  // ...
+
+  @method vote(...) {
+    this.network.timestamp.assertBetween(startDate, endDate);
+    // ...
+  }
+}
+```
+
+A more refined example could store the current start date in an on-chain state variable, which can be reset by some process which is also encoded by the zkApp.
+
+In addition to using a predefined range, you can also construct a range that depends on another variable, such as the current time. For example, a DEX with an order book could support orders that expire after an hour. So, the range should be `[now, now + 1h]`. This works just as you'd expect:
```

Comment:
Unfortunately, this example makes no sense. Can be deleted before merging, or replaced with one that makes sense